care/facility/tests/test_apis/test_ambulance_api.py

```diff
@@ -0,0 +1,285 @@
+            "id": ambulance.id,
+            "drivers": [self._get_driver_representation(driver) for driver in ambulance.drivers],
+            "primary_district_object": self._get_district_object_representation(ambulance.primary_district),
+            "secondary_district_object": self._get_district_object_representation(ambulance.secondary_district),
+            "third_district_object": self._get_district_object_representation(ambulance.third_district),
+            "created_date": mock_equal,
+            "modified_date": mock_equal,
+            "deleted": ambulance.deleted,
+            "vehicle_number": ambulance.vehicle_number,
+            "owner_name": ambulance.owner_name,
+            "owner_phone_number": ambulance.owner_phone_number,
+            "owner_has_smart_phone": ambulance.owner_has_smart_phone,
+            "has_oxygen": ambulance.has_oxygen,
+            "has_ventilator": ambulance.has_ventilator,
+            "has_suction_machine": ambulance.has_suction_machine,
+            "has_defibrillator": ambulance.has_defibrillator,
+            "insurance_valid_till_year": ambulance.insurance_valid_till_year,
+            "ambulance_type": ambulance.ambulance_type,
+            "price_per_km": ambulance.price_per_km,
+            "has_free_service": ambulance.has_free_service,
+            "primary_district": ambulance.primary_district.id,
+            "secondary_district": getattr(ambulance.secondary_district, "id", None),
+            "third_district": getattr(ambulance.third_district, "id", None),
+        }
+
+    def get_detail_representation(self, ambulance: Any = None) -> dict:
+        if isinstance(ambulance, dict):
+            # can't assign district id here
+            ambulance.update(primary_district=self.district)
+            ambulance.pop("drivers", "[]")
+
+            ambulance = Ambulance(**ambulance)
+
+        list_repr = self.get_list_representation(ambulance)
+        detail_repr = list_repr.copy()
+        # no changes in list repr and detail repr, if there are only those may be updated here.
+        detail_repr.update({})
+        return detail_repr
+
+    def _get_driver_representation(self, driver: Ambulance.drivers or dict):
+        """
+        Returns ambulance driver representation for ambulance
+
+        Returns:
+            dict
+                Representing attributes of the driver
+
+        Params:
+            driver: Ambulance.drivers or dict
+        """
+        if isinstance(driver, dict):
+            return {
+                "id": mock_equal,
+                "name": driver["name"],
+                "phone_number": driver["phone_number"],
+                "has_smart_phone": driver["has_smart_phone"],
+            }
+        return {
+            "id": driver.id,
+            "name": driver.name,
+            "phone_number": driver.phone_number,
+            "has_smart_phone": driver.has_smart_phone,
+        }
+
+    def _get_district_object_representation(self, district):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5340830)